torchtune/datasets/_grammar.py

```diff
@@ -4,17 +4,54 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-from torchtune.datasets._instruct import instruct_dataset, InstructDataset
-from torchtune.modules.tokenizers import ModelTokenizer
+
+from typing import Any, Dict, Mapping, Optional
+
+from torch.utils.data import Dataset
+from torchtune.data import Message
+from torchtune.data._templates import GrammarErrorCorrectionTemplate
+from torchtune.datasets._finetune import FinetuneDataset
+from torchtune.datasets._packed import PackedDataset
+from torchtune.modules.transforms import Transform
+
+
+class GrammarMessages(Transform):
+    def __init__(
+        self, train_on_input: bool = False, column_map: Optional[Dict[str, str]] = None
+    ):
+        self.train_on_input = train_on_input
+        self.column_map = column_map
+        self.template = GrammarErrorCorrectionTemplate()
```

Comment:
I was initially leaning towards that, but in some cases the prompt templates may be model specific (e.g. MistralChatFormat, Llama2ChatFormat, ChatMLFormat) so I decided to make them standalone components. Also, prompt templates are more task specific rather than dataset specific. Actually, I think making this the default template but also changeable is preferrable.